proc/threads.c

```diff
@@ -787,68 +791,82 @@ int proc_threadCreate(process_t *process, void (*start)(void *), unsigned int *i
 {
 	thread_t *t;
 	spinlock_ctx_t sc;
+	int err = EOK;
 
-	if (priority >= sizeof(threads_common.ready) / sizeof(thread_t *))
-		return -EINVAL;
+	if (priority >= sizeof(threads_common.ready) / sizeof(thread_t *)) {
+		err = -EINVAL;
+	}
 
-	if ((t = vm_kmalloc(sizeof(thread_t))) == NULL)
-		return -ENOMEM;
+	if ((t = vm_kmalloc(sizeof(thread_t))) == NULL) {
+		err = -ENOMEM;
+	}
```

Comment:
This code refactor is out-of-scope of this PR (and not ideal too, if we fail here there is no need for stack allocation below, also we should extract vm_kmalloc call from if to be MISRA friendly).